### ISSUE TITLE
The SDK crashes in -[ANGDPRSettings getDeviceAccessConsent] when using TCF 2.0 purpose consent strings

### ISSUE THREAD

**Issue by ErikLuimes** [appnexus/mobile-sdk-ios#51]
This crash happens when using TCF 2.0 and the purpose consent string is empty. 

You get an index out of bounds because a substring is created from an empty string. This happens because the if statement evaluates the string (as opposed to its length) to be greater than zero.

https://github.com/appnexus/mobile-sdk-ios/blob/c52bc9202088957a0a1208c05ee7f30d74f4bcba/sdk/sourcefiles/internal/config/ANGDPRSettings.m#L117-L119

should be:
```   
if(purposeConsents.length > 0){
    return [purposeConsents substringToIndex:1];
}
```

**Comment by po-dpg** [comment]
I would leave the nil check in it before the empty string check

**Comment by ppuviarasu** [comment]
Implemented & released with v7.4.1.
